tests/models/test_study.py

```diff
@@ -0,0 +1,482 @@
+import datetime
+import json
+
+import pytest
+import pytz
+from sqlalchemy import exc
+
+from gdc_ng_models.models import studyrule
+
+
+def test_study_rule__defaults_unique_ids(create_study_rule_db, db_session):
+    db_session.add(studyrule.StudyRule(name="first"))
+    db_session.add(studyrule.StudyRule(name="second"))
+    db_session.commit()
+
+    first = db_session.query(studyrule.StudyRule).filter(studyrule.StudyRule.name == "first").first()
+    second = db_session.query(studyrule.StudyRule).filter(studyrule.StudyRule.name == "second").first()
+    assert first.id is not None
+    assert second.id is not None
+    assert first.id != second.id
+
+
+def test_study_rule__unique_id(create_study_rule_db, db_session):
+    db_session.add(studyrule.StudyRule(id=1618, name="first"))
+    db_session.commit()
+
+    with pytest.raises(exc.IntegrityError, match=r"study_rule_pk"):
+        db_session.add(studyrule.StudyRule(id=1618, name="second"))
+        db_session.commit()
+
+
+def test_study_rule__name_must_be_defined(create_study_rule_db, db_session):
+    with pytest.raises(exc.IntegrityError, match=r"violates not-null constraint"):
+        db_session.add(studyrule.StudyRule(id=1618))
+        db_session.commit()
+
+
+def test_study_rule__unique_name(create_study_rule_db, db_session):
+    db_session.add(studyrule.StudyRule(name="dupe"))
+    db_session.commit()
+
+    with pytest.raises(exc.IntegrityError, match=r"study_rule_name_idx"):
+        db_session.add(studyrule.StudyRule(name="dupe"))
+        db_session.commit()
+
+
+def test_study_rule__defaults_created_date(create_study_rule_db, db_session):
+    db_session.add(studyrule.StudyRule(name="asdf"))
+    db_session.commit()
+    s = db_session.query(studyrule.StudyRule).filter(studyrule.StudyRule.name == "asdf").first()
+    assert s.created_date is not None
+
+
+def test_study_rule__defaults_updated_date(create_study_rule_db, db_session):
+    db_session.add(studyrule.StudyRule(name="asdf"))
+    db_session.commit()
+    s = db_session.query(studyrule.StudyRule).filter(studyrule.StudyRule.name == "asdf").first()
+    assert s.updated_date is not None
+
+
+def test_study_rule__updates_updated_date(create_study_rule_db, db_session):
+    db_session.add(studyrule.StudyRule(id=1618, name="asdf"))
+    db_session.commit()
+
+    db_session.query(studyrule.StudyRule).filter(studyrule.StudyRule.id == 1618).update({"name": "updated"})
+    db_session.commit()
+
+    updated = db_session.query(studyrule.StudyRule).filter(studyrule.StudyRule.id == 1618).first()
+    assert updated.updated_date > updated.created_date
```

Comment:
To make really sure that the `updated_date` is bumped on update, and isn't just coincidentally set to a greater value than the `created_date` when the row is first inserted, I think it'd be a good idea to grab the `updated_date` before and after the update, and compare the two `updated_date`s.